snap/build.sh

```diff
@@ -16,22 +16,33 @@ if [ ! -z "$JENKINS_URL" ]; then
         echo "I seem to be running on Jenkins, but there's not a snap store login file..." 
     fi
 
-    # check if this is a release job or not, if it is set the corresponding env var
+    # figure out what kind of job this is using $JOB_NAME and simplify that 
+    # into $JOB_TYPE
+    JOB_TYPE="build"
     if [[ "$JOB_NAME" =~ edgex-go-snap-.*-stage-snap.* ]]; then
-        IS_RELEASE_JOB="YES"
-    else
-        IS_RELEASE_JOB="NO"
+        JOB_TYPE="stage"
+    elif [[ "$JOB_NAME" =~ edgex-go-snap-.*-release-snap.* ]]; then
+        JOB_TYPE="release"
     fi
 fi
 
-# build the container image - providing the relevant architecture base image
-docker build -t edgex-snap-builder:latest -f ${SCRIPT_DIR}/Dockerfile.build $GIT_ROOT
+# build the container image - providing the relevant architecture we're on
+# to determine which snap arch to download in the docker container
+case $(arch) in 
+    x86_64)
+        arch="amd64";;
+    aarch64)
+        arch="arm64";;
+    arm*)
+        arch="armhf";;
```

Comment:
If **armhf** isn't yet supported, why include it here?

Reply:
Because I think eventually we will build the snap for armhf and it's easier to handle these changes now rather than go back and add them later